app/src/main/java/io/gnosis/safe/Tracker.kt

```diff
@@ -0,0 +1,76 @@
+package io.gnosis.safe
+
+import android.content.Context
+import android.os.Bundle
+import com.google.firebase.analytics.FirebaseAnalytics
+import com.google.firebase.crashlytics.FirebaseCrashlytics
+import io.gnosis.safe.di.ApplicationContext
+import javax.inject.Inject
+
+class Tracker @Inject constructor(@ApplicationContext context: Context) {
```

Comment:
Should this be marked as `@Singleton` ?

Reply:
It's provided as a singleton in a module. Is @Singleton annotation necessary? Or you mean it would improve readability?

Reply:
The annotation changes the generated code and in some cases does some checking. Easiest to see that if you look at the generated code. The singleton should be cached and only be created once, while the unscoped one will be used like a factory and be created each time again

Reply:
☝️ exactly what Richard said. Particularly, because it could be handy as well eventually to have a reference to `this` in a `companion object` so that we can track in viewHolders when constructor or member injection is not posible.